### PR TITLE
2572119 data convert action

### PR DESCRIPTION
**Pull request by ndeet**
See d.o. issue https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/RulesAction/DataConvert.php [hunk 1]
```diff
@@ -43,13 +43,15 @@ class DataConvert extends RulesActionBase {
...
+   * @param string $rounding_behavior
+   *   The behaviour for rounding.
    */
-  public function execute() {
-    $value = $this->getContextValue('value');
-
-    $target_type = $this->getContextValue('target_type');
-    $rounding_behavior = $this->getContextValue('rounding_behavior');
-
+  public function doExecute($value, $target_type, $rounding_behavior) {
```
Comment (klausi) [inline src/Plugin/RulesAction/DataConvert.php]: should be protected

File: src/Plugin/RulesAction/DataConvert.php [hunk 2]
```diff
@@ -43,13 +43,15 @@ class DataConvert extends RulesActionBase {
 
   /**
    * Executes the plugin.
+   *
+   * @param mixed $value
+   *   The input value.
+   * @param string $target_type
+   *   The target for the action.
```
Comment (klausi) [inline src/Plugin/RulesAction/DataConvert.php]: should be the "The target type the value should be converted into."

**Comment by ndeet** [comment]
Fixed.

File: src/Plugin/RulesAction/DataConvert.php [hunk 3]
```diff
@@ -43,13 +43,15 @@ class DataConvert extends RulesActionBase {
...
+   * @param string $rounding_behavior
+   *   The target type the value should be converted into.
```
Comment (klausi) [inline src/Plugin/RulesAction/DataConvert.php:50]: nope, this is the rounding behavior.

**Comment by ndeet** [comment]
Right, talking and coding does not work. Fixed.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**